### DISCUSSION TITLE
Scheduled job is being processed immediately

### DISCUSSION THREAD

**Discussion by mbajur** [bensheldon/good_job#1563] [Q&A]
Hey! I'm trying to test if one of service objects of mine is rescheduling jobs properly. I simplified my test case to demonstrate my issue:

```ruby
around do |ex|
  original_adapter = ProcessJob.queue_adapter
  queue_adapter = :good_job
  ex.run
  ProcessJob.queue_adapter = original_adapter
end

it "reschedules the job" do
  job_id = ProcessJob.set(at: 2.days.from_now).perform_later.job_id
  pp GoodJob::Job.find(job_id)
end
```

After inspecting the `GoodJob::Job` found, i can see that this job has been immidiatelly processed, not respecting the schedule time provided. I can see in the readme that i can set `Rails.application.config.good_job.inline_execution_respects_schedule = true` to make good_job respect that setting but it doesn't help, my job is still being processed instantly. Any ideas what am i doing wrong?

I'm on good_job 3.99.0.

Thank you!

**Comment by bensheldon** [comment]
That's weird! When you inspect the `GoodJob::Job` instance, is `scheduled_at` set and in the future?

I would expect the jobs not to be executed inline because of this line:

https://github.com/bensheldon/good_job/blob/fc3add15e7cf8c051732082261bdc501dd8ebce1/lib/good_job/adapter.rb#L94-L95

btw, `inline_execution_respects_schedule = true` is the default as of GoodJob v3

**Reply by mbajur** [reply]
It is not set in the future, `scheduled_at` is set to the same value as creation date (`Time.current`)
